### PR TITLE
[docs] Set "hostNetwork: true" for toolbox

### LINKED ISSUE #2021: "rbd map" hangs in toolbox
As stated in the [docs](https://rook.io/docs/rook/v0.8/direct-tools.html), `rbd map` hangs (while still succeeding) when running it in the toolbox:
```sh
# Map the rbd device. If this command appears to hang, ctrl-c and check if it actually completed successfully by continuing.
rbd map replicapool/test
```

Because I was a bit annoyed by this, I decided to look into it. With a bit of help from #ceph-devel on IRC, I found that, after mapping the block image, "rbd map" [waits for an udev add event](https://github.com/ceph/ceph/blob/master/src/krbd.cc#L182-L251) for the /dev/rbd*X* block device. Using `udev monitor` inside the toolbox I noticed it didn't receive those events (while `udev monitor` on the host was receiving them). Googling around, I [found out](https://dummdida.tumblr.com/post/121087781445/re-udev-events-in-a-container) that the only way to get a container to receive udev events is to run the container with `--net=host`. Changing https://github.com/rook/rook/blame/master/Documentation/toolbox.md#L47 to `hostNetwork: true` fixed the hanging issue.

So, my suggestion is to change `hostNetwork: true` in the toolbox documentation (maybe with a comment that setting it to `false` will make `rbd map` hang), and refine the already existing hang comment. If you agree, I can create a PR.

### PR DESCRIPTION
**Pull request by lbogdan**
Signed-off-by: Bogdan Luca <luca.bogdan@gmail.com>

Resolves #2021 

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [x] `make vendor` does not cause changes.

[skip ci]

### CODE REVIEW

**Comment by galexrt** [comment]
Could you also update the toolbox yaml in the `cluster/` directory.

**Comment by lbogdan** [comment]
@galexrt Updated, sorry for missing that!

File: Documentation/toolbox.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ spec:
         name: libmodules
       - name: mon-endpoint-volume
         mountPath: /etc/rook
-  hostNetwork: false
+  hostNetwork: true # if false, the "crd map" command will hang, see https://github.com/rook/rook/issues/2021
```
Comment (galexrt) [inline Documentation/toolbox.md]: Shouldn't `crd` be `rbd`?
Also could you move the comment on a new line before the `hostNetwork` line to align it with other yaml comments (see yamls in `cluster/` directories).

File: cluster/examples/kubernetes/ceph/toolbox.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ spec:
         name: libmodules
       - name: mon-endpoint-volume
         mountPath: /etc/rook
-  hostNetwork: false
+  hostNetwork: true
```
Comment (galexrt) [inline cluster/examples/kubernetes/ceph/toolbox.yaml]: Could you also add a comment here but above the `hostNetwork` line.

File: Documentation/toolbox.md (hunk 1 above)
Comment (lbogdan) [reply Documentation/toolbox.md]: Fixed typo and moved comment above.

File: cluster/examples/kubernetes/ceph/toolbox.yaml (hunk 2 above)
Comment (lbogdan) [reply cluster/examples/kubernetes/ceph/toolbox.yaml]: Added.

**Pull request merged**